UCSD-E4E/Pyrenote
# Clip retirement

Created a clip retirement system! Based on the precision between users, Pyrenote will automatically retire clips with high degrees of overlap. This allows users to annotate previously poorly labeled clips in the attempt in improving metrics

Added
- New system for recommending clips!
   - IOU based clip retirement based on number of annotators and/or precision of their overlaps
   - Added controls to iou retirement to edit projects forms

Testing:
- Create a new project, be sure to enable iou via Enable Experimental Quality Control in the rename menu and set desired max_user count and confidence threshold for retirements. (for testing, it helps to turn on example project to speed up testing)
- Label the data with an admin account
- Go to a new browser (recommended) and create a new account
- Observe the clips you annotated as an admin appear in pending list in data with the new user. For the admin users, they will appeared as completed
- With the new user, label some clips as you did before, and then label other clips poorly
- Create a second new user, and open the project. Observe the clips you labeled well do not appear in pending, the clips labeled poorly will appear for this second users
- In the admin account, try changing the max_user count and threshold limits. Observe changes for the second new user in the pending list
- Repeat this process until all clips are retired

TODO
- ~~Check to make sure that the quick load data button and other data viewing types (all, marked for review, etc) are working normally~~
- ~~Add a retired clips view on frontend in the data portal~~
    -  ~~Make sure this is limited to admins only~~
- ~~add a data column in sql tables that denotes which clips are considered "retired" for data results purposes~~
- ~~Add counted label to csv~~

BUGS
- Make sure that on systems without IOU metrics turned on, the data appears in the data portal
